contracts/multisig/src/multisig_propose.rs

```diff
@@ -190,6 +201,12 @@ pub trait MultisigProposeModule: crate::multisig_state::MultisigStateModule {
                     own_shard == other_sc_shard,
                     "All transfer exec must be to the same shard"
                 );
+            } else if let Action::SendTransferExecuteEsdt(call_data) = &action {
```

Comment:
Can we please refactor the propose_batch function? It's starting to become crowded. We can remove some conditionals into small functions.